### PR TITLE
NN-3723: your completed reports page

### PR DESCRIPTION
**Pull request by richardpopple**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/RestResponsePage.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.pagination
```
Comment (richardpopple) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/RestResponsePage.kt:1]: This class is required so jackson can deserialise Spring Page objects

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
+  )
+  return params.map { it.key + "=" + it.value }
 }
 
-data class PageRequest(
-  val pageNumber: Long,
-  val pageSize: Long,
-  val firstPage: Long = 1
-)
+fun getSortUrlParameters(pageable: Pageable): List<String> {
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt:15]: private?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt [hunk 3]
```diff
@@ -1,22 +1,17 @@
...
-) {
+import org.springframework.data.domain.Pageable
 
-  fun <R> map(transform: (T) -> R): PageResponse<R> = PageResponse(this.pageNumber, this.pageSize, this.totalResults, this.results.map(transform), this.firstPage)
+fun getPageableUrlParameters(pageable: Pageable): String = getPageUrlParameters(pageable).plus(getSortUrlParameters(pageable)).joinToString("&")
 
-  companion object {
-    fun <T> emptyPageRequest(pageRequest: PageRequest): PageResponse<T> = PageResponse(pageRequest.pageNumber, pageRequest.pageSize, 0, emptyList(), pageRequest.firstPage)
-  }
+fun getPageUrlParameters(pageable: Pageable): List<String> {
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt:7]: private?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt [hunk 4]
```diff
@@ -1,22 +1,17 @@
 package uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.pagination
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt:1]: Really the classes in this package should be in the gateway package (or under it), as it is only used there

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/RestResponsePage.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.JsonNode
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+
+class RestResponsePage<T> : PageImpl<T> {
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/RestResponsePage.kt:10]: Why do we need this? It seems to just pass everything thru to PageImpl?

**Review by sp-watson — APPROVED** [review]
Some comments to consider, but can be sorted out some other time

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/controllers/ReportedAdjudicationController.kt [hunk 6]
```diff
@@ -45,25 +39,14 @@ class ReportedAdjudicationController {
...
-  }
-
-  @GetMapping("/my")
-  fun getMyReportedAdjudications(): MyReportedAdjudicationsResponse {
-    val myReportedAdjudications = reportedAdjudicationService.getMyReportedAdjudications()
-
-    return MyReportedAdjudicationsResponse(
-      reportedAdjudications = myReportedAdjudications
-    )
+  @ApiOperation(value = "Get my reported adjudications for caseload")
```
[the remark this answers is not in the document]
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/controllers/ReportedAdjudicationController.kt:42]: Cheers - I'll make the changes in the next request

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt (hunk 4 above)
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt:1]: I'll refactor

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt (hunk 3 above)
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt:7]: yes

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt (hunk 2 above)
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/Pagination.kt:15]: yes

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/RestResponsePage.kt (hunk 5 above)
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/pagination/RestResponsePage.kt:10]: Yeah - it does - but the Spring instance (PageImpl) can't be deserialised by jackson as there is no default constructor on it - hence the need for this :(

**Pull request merged**